### PR TITLE
chore: update app setup and webpack

### PR DESCRIPTION
**Pull request by donni106**
- updated rails to 6.1.4
- removed every vue occurrence as we do not use it in the entire app
  - this seems to be some relict from initial setup
- upgraded some node packages to more recent versions
- fixed file name of .nvmrc and increased version
- integrated moment as package instead of from cdn
- renamed application to get rid of this weird "jsonform_cms"
- reverted assets.rb as we do not need the added precompile path
- removed babel loose mode for plugin-proposal-class-properties as it produced too much warnings
  - https://2ality.com/2015/12/babel6-loose-mode.html
- autolinted some files
- gitignored production key

### CODE REVIEW

**Comment by donni106** [comment]
**PROBLEM** something does not work with datatables, so that styling is missing. I have not found out the reason after hours of versioning, reverting and stuff...

**Comment by donni106** [comment]
> **PROBLEM** something does not work with datatables, so that styling is missing. I have not found out the reason after hours of versioning, reverting and stuff...

done! we are not allowed to update `datatables.net` from `^1.10.19` to `1.11.3` right now!

File: config/initializers/assets.rb [hunk 1]
```diff
@@ -11,4 +11,4 @@
 # Precompile additional assets.
 # application.js, application.css, and all non-JS/CSS in the app/assets
 # folder are already added.
-Rails.application.config.assets.precompile += %w[ckeditor/config.js]
```
Comment (marcometz) [inline config/initializers/assets.rb:14]: ckeditor/config.js hier raus aus dem Precompile?

**Review by marcometz — APPROVED** [review]

File: config/initializers/assets.rb (hunk 1 above)
Comment (donni106) [reply config/initializers/assets.rb:14]: siehe commit message

> - reverted assets.rb as we do not need the added precompile path

wird nicht benötigt. den pfad gibts/gabs gar nicht.

**Pull request merged**